share/harvest/base.py

```diff
@@ -115,7 +114,7 @@ def fetch_date(self, date: datetime.date, **kwargs):
         """
         return self.fetch_date_range(date - datetime.timedelta(days=1), date, **kwargs)
 
-    def fetch_date_range(self, start, end, limit=None, **kwargs):
+    def fetch_date_range(self, start, end, limit=None):
```

Comment:
Right now this is only used for `set_spec`, which is already in `harvester_kwargs`, so the OAI harvester needs extra logic to manually handle the override (`set_spec = set_spec or self.set_spec`). I think the BaseHarvester API would be better and easier to work with if harvester-specific kwargs enter in only one place (`__init__`, stored in `self.kwargs` by default), not two.

When using a harvester, it's just a change from
```
harvester = config.get_harvester()
harvester.fetch(set_spec='my-crazy-setspec')
```
to
```
harvester = config.get_harvester(set_spec='my-crazy-setspec')
harvester.fetch()
```
which seems like no sacrifice. It'd also be nice to guarantee that any kwargs you can provide manually can also be specified in `harvester_kwargs`, and vice versa.